### DISCUSSION TITLE
Is it possible to create a read-only pdf?

### DISCUSSION THREAD

**Discussion by cannonegiuseppe** [Hopding/pdf-lib#1153] [General]
I am trying to create a pdf with sensitive data and therefore I would like it to be read-only without the possibility to make changes to it. I have read the documentation but I found the option to make a form read-only, but I couldn't find anything to make the whole page read-only. 
If you know how to do please help me, thanks.

**Comment by crockwave** [comment]
When you look at the [PDF 1.7 Specification](https://opensource.adobe.com/dc-acrobat-sdk-docs/standards/pdfstandards/pdf/PDF32000_2008.pdf) pdf file in the current Microsoft Edge browser, the built-in PDF reader identifies the following document permissions, which indicates that the document is read-only.

Look at section 7.6.3.1 in that document for setting general permissions

![image](https://user-images.githubusercontent.com/11214905/160019322-bba0d4f4-76bd-4d21-bb87-b6f0e114282d.png)

**Comment by crockwave** [comment]
https://stackoverflow.com/questions/65518525/how-to-convert-editable-pdf-to-non-editable-pdf-in-php-nodejs
